src/com/xilinx/rapidwright/design/DesignTools.java

```diff
@@ -2460,23 +2539,39 @@ public static List<String> getAllRoutedSitePinsFromPhysicalPin(Cell cell, Net ne
      */
     public static void createMissingSitePinInsts(Design design) {
         EDIFNetlist netlist = design.getNetlist();
-        for (Net net : design.getNets()) {
-            if (net.isUsedNet()) {
-                continue;
-            }
-            EDIFHierNet ehn = net.getLogicalHierNet();
-            EDIFHierNet parentEhn = (ehn != null) ? netlist.getParentNet(ehn) : null;
-            if (parentEhn != null && !parentEhn.equals(ehn)) {
-                Net parentNet = design.getNet(parentEhn.getHierarchicalNetName());
-                if (parentNet != null) {
-                    // 'net' is not a parent net (which normally causes createMissingSitePinInsts(Design, Net)
-                    // to analyze its parent net) but that parent net also exist in the design and has been/
-                    // will be analyzed in due course, so skip doing so here
-                    continue;
+        Map<SiteInst, Map<Net, List<String>>> siteInstToNetSiteWiresMap = DesignTools.getSiteInstToNetSiteWiresMap(design);
+        final CountUpDownLatch threadsOutstanding = new CountUpDownLatch();
+        int numNets = design.getNets().size();
+        int numJobs = ParallelismTools.maxParallelism() * 100;
```

Comment:
We should describe in a comment why 100 (experimentally was the best performing...)